usefulstuff.c

```diff
@@ -66,7 +66,7 @@ char* xdebug_fd_read_line_delim(int socketfd, fd_buf *context, int type, unsigne
 		context->buffer_size = 0;
 	}
 
-	while (context->buffer_size < 1 || context->buffer[context->buffer_size - 1] != delim) {
+	while((ptr=memchr(context->buffer, delim, context->buffer_size))==NULL){
```

Comment:
Please stick to the coding standard with a space after the `while`, with spaces around the `=`, and `==`, and a space before the `{` (at the end of the line).